src/controllers/student.controls.js

```diff
@@ -9,68 +9,84 @@ const CourseRepository = require('../repositories/courses.repository');
  * @param {Next} next - Callback function
  * @returns {Object} - Success response in JSON
  */
-const studentRepository = new StudentRepository();
-const courseRepository = new CourseRepository();
+
 module.exports = {
-    addStudent: async(req, res, next)=>{
-        try {
-            const newStudent = await studentRepository.addStudent(req.body);
-            res.status(201).json(newStudent);
-        } catch (error) {
-            res.status(400).json(error);
-        }
-    },
-    getStudents: async(req, res, next)=>{
-        try {
-            const students = await studentRepository.getAllStudent();
-            res.status(200).json(students);
-        } catch (error) {
-            res.status(400).json(error);
-        }
-    },
-    getStudentById: async(req, res, next)=>{
-        try {
-            const student = await studentRepository.getStudentById(req.params.id);
-                res.status(200).json(student);
-        } catch (error) {
-            next(error);
-        }
-    },
-    updateStudent: async(req, res, next)=>{
-        try {
-            const student = await studentRepository.updateStudent(req.body, req.params.id);
-            res.status(201).json(student);
-        } catch (error) {
-            res.status(400).json(error);
-        }
-    },
-    deleteStudentById: async(req, res, next)=>{
-        try {
-            await studentRepository.deleteStudentById(req.params.id);
-            res.status(204).end();
-        } catch (error) {
-            res.status(400).json(error);
-        }
-    },
-    takeCouse: async(req, res, next)=>{
-        try{
-            const course = await courseRepository.getCourseByCode(req.params.code);
-            if(course.code){
-                const student = await studentRepository.takeCouse(req.params.id, course);
-                res.status(202).json(student);
-            }
-        }
-        catch(error){
-            res.status(400).json(error);
-        }
-    },
-    dropCouse: async(req, res, next)=>{
-        try{
-            const result = await studentRepository.dropCouse(req.params.id, req.params.code);
-            res.status(202).json(result);
-        }
-        catch(error){
-            res.status(400).json(error);
-        }
+  addStudent: async (req, res, next)=>{
+    const id = generateId();
+    const newStudent = new Model({
+      studentId: id,
```

Comment:
question. Every document will be created with a unique `_id`. Is `studentId` a needed?
You will already have a unique id for each student with `student._id`

Reply:
I want an Id that is also easy to remember

Reply:
HM, okay to leave as-is. But I do think this is a poor design. You don't think `student._id` is easy to remember? Every single mongo document has a unique `_id` by default. And most developers will know this and assume that when you say `studentId` you mean the `_id` of the `student` document